source/upgrade.txt

```diff
@@ -161,6 +174,10 @@ The v4.0 driver introduces the following breaking changes to the ABI:
   library equivalents. For context, see the :ref:`Build System Breaking Changes
   <version-4.0-build-system-breaking>` section.
 
+- Removes deprecated ``utf8`` types and functions that have ``string`` equivalents.
```

Comment:
```suggestion
- Removes deprecated ``utf8`` functions that have ``string`` equivalents.
```
Types are not part of the ABI. Only variables and functions are part of the ABI.